### ISSUE TITLE
Is there a way to conditionally Skip same API path which is added to mock

### ISSUE THREAD

**Issue by NikJ87** [miragejs/miragejs#1050]
Hi Team,

I'm looking for a way to following but didn't get any success:

`
this.get('/books', (schema, req) => {
  if (req.queryParam.author)
   ```
this.passthrough(req => {
      return req.queryParam.author === 'Agatha'
   })
```
else 
    new Response(200, {some-headers}, {data: books})
})
`

Thanks
Nik

**Comment by cah-briangantzler** [comment]
I do not believe you can do this. The books URL is intercepted. When `passthrough` is used it is defining what routes to allow to pass through, not actually perform the route call. 

I do see that this could be accomplished with a PR to pretender. Currently this code https://github.com/pretenderjs/pretender/blob/master/src/create-passthrough.ts#L19 is embedded in the creation of the passthrough interceptor. If this code were extracted, the interceptor could call it and you could to, providing the needed information was available on the `req` param.
